File: quartz-energy/src/backend/QuartzEnergy.Common.Services/Models/Statistics/NoRequestStatistics.cs
namespace QuartzEnergy.Common.Services.Models.Statistics
{
    using System.Collections.Generic;

    public class NoRequestStatistics<TData> : Statistics<EmptyStatisticsRequest, TData>
    {
        public NoRequestStatistics(IEnumerable<StatisticsItem<TData>> items)
            : base(new EmptyStatisticsRequest(), items)
        {
        }
    }
}

File: quartz-energy/src/frontend/src/app/common/models/statistics/no-request-statistics.ts
import { StatisticsItem } from './statistics-item';
import { EmptyStatisticsRequest } from './empty-statistics-request';
import { Statistics } from './statistics';

export class NoRequestStatistics<TData> extends Statistics<EmptyStatisticsRequest, TData> {
    constructor(
        public items: StatisticsItem<TData>[]
    ) {
        super(new EmptyStatisticsRequest(), items);
    }
}

